static/js/settings.js

```diff
@@ -0,0 +1,63 @@
+(function() {
+	"use strict";
+
+    // TODO Find a way to make this reusable across the app?
+    function apiRequest(path, options) {
+        const cookieString = typeof document.cookie === "string" ? document.cookie : "";
+        const cookieStringArray = cookieString
+            .split(";")
+            .map(individualCookieString => individualCookieString.split("="))
+            .map(([cookieKey, cookieValue]) => [cookieKey.trim(), cookieValue.trim()]);
+        // Looks like the `argsIgnorePattern` option for ESLint doesn't like array destructuring:
+        // eslint-disable-next-line no-unused-vars
+        const [_csrfCookieKey, csrfCookieValue] = cookieStringArray.find(([cookieKey, _cookieValue]) => cookieKey === "csrftoken");
+        const headers = new Headers(options ? options.headers : undefined);
+        headers.set("X-CSRFToken", csrfCookieValue);
+        headers.set("Content-Type", "application/json");
+        return fetch(
+            `/api/v1${path}`,
+            {
+                mode: "same-origin",
+                ...options,
+                headers: headers,
+            },
+        );
+    }
+
+    const settingsForm = document.querySelector("#settings-form");
```

Comment:
Ha, so that's why I defaulted to first, but then I saw the user of `querySelectorAll` elsewhere in the codebase instead of `getElementsByTagName` (for example) and I looked into it a bit. Turns out the latter returns an `HTMLCollection` (which stays up to date with DOM changes) whereas the former just returns a static `NodeList`, so I figured that that was probably for performance reasons too.

But of course, that doesn't carry over to `getElementById` (which just returns a single element in the first place).

All of which is a long-winded way of saying: 652f38ead0ddf7dc1fa2ac140208fdc0bf243059.